### PR TITLE
Elasticsearch pipeline

### PR DESCRIPTION
**Pull request by yinzhigang**
add support of elasticsearch pipeline: https://www.elastic.co/guide/en/elasticsearch/reference/current/ingest.html

It's powerful pre-process documents before indexing, support like JSON text #134 , geo point #218 and more.

require elasticsearch version >= 5.0

### CODE REVIEW

File: river/sync.go [hunk 1]
```diff
@@ -242,13 +242,21 @@ func (r *River) makeUpdateRequest(rule *Rule, rows [][]interface{}) ([]*elastic.
...
+			req = &elastic.BulkRequest{Index: rule.Index, Type: rule.Type, ID: afterID, Parent: afterParentID, Pipeline: rule.Pipeline}
 			r.makeInsertReqData(req, rule, rows[i+1])
 
 			r.st.DeleteNum.Add(1)
 			r.st.InsertNum.Add(1)
 		} else {
-			r.makeUpdateReqData(req, rule, rows[i], rows[i+1])
+			if len(rule.Pipeline) > 0 {
+				// Pipelines can only be specified on index action
+				r.makeInsertReqData(req, rule, rows[i+1])
```
Comment (siddontang) [inline river/sync.go]: Why use insert here? Do we need to send all columns data here?

Comment (yinzhigang) [reply river/sync.go]: Yes, Different from update action, index action need all columns data. Otherwise ES will delete non-existent filds

**Comment by siddontang** [comment]
LGTM

**Comment by M2shad0w** [comment]
@yinzhigang how to config it  in rule file ?

**Comment by SuperNoobTao** [comment]
@yinzhigang  hello , it seems that json processor can not deal with json array. I have a json array in the mysql column. do you have any ideas ?

**Comment by purevirtual** [comment]
> @yinzhigang hello , it seems that json processor can not deal with json array. I have a json array in the mysql column. do you have any ideas ?

I guess you can try "Script Processor"

**Pull request merged**